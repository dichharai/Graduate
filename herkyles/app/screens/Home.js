import React, { Component } from "react"; 
import { View, Text, StyleSheet, StatusBar, Linking} from "react-native"; 
import { StackNavigator } from "react-navigation"; 
 
import Container from "../components/Container/Container";
import LinkBtns from "../components/Buttons/LinkBtns/LinkBtns"; 
import Logo from "../components/Logo/Logo";
import {firebaseApp} from '../../db/DbConfig';
 
var buttonLink = "";


class Home extends Component{
    static navigationOptions = {
        title: "Home", 
        headerStyle: {
            backgroundColor: "#000000",
        }, 
          headerTintColor: '#facf33',
    };

    

    handleGymStatsPress = () => {
        this.props.navigation.navigate("GymStats"); 
        console.log("handle gym stats press"); 
    }

    handleGroupPress = () => {
        Linking.openURL(this.state.link);
        console.log("handle group press"); 
    }

    handleCodeScannerPress = () => {
        this.props.navigation.navigate("CodeScanner");
        console.log("handle code scanner press");
    }

    handleLoginPress = () => {
        this.props.navigation.navigate("Login"); 
        console.log("handle login press"); 
    }

    handleCodeScannerPress = () => {
        this.props.navigation.navigate("CodeScanner"); 
        console.log("handle code scanner press"); 
    }

    constructor(props){
        super(props);

        var mainRef = firebaseApp.database().ref("groupFitnessScheduleLink")
        
        mainRef.once("value").then(function(dataSnapshot) {
            buttonLink = dataSnapshot.val();
        })
    }
    render(){
        return (
            <Container>
                
                <StatusBar transclucent={false} barStyle="light-content"/>
                <Logo/>
                <LinkBtns
                    text="Gyms"
                    onPress={this.handleGymStatsPress}
                />
                <LinkBtns
                    text="Group Fitness Schedule"
                    onPress={()=> {Linking.openURL(buttonLink)}}
                />
                <LinkBtns
                    text="QR Code Scanner"
                    onPress={this.handleCodeScannerPress}
                />
                <LinkBtns
                    text="Login"
                    onPress={this.handleLoginPress}
                />
            </Container>
          
        );
    }
}
 


export default Home; 