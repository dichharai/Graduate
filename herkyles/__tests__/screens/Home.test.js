import React from "react";
import Home from "../../app/screens/Home";
import Enzyme, {shallow} from 'enzyme'; 
import Adapter from 'enzyme-adapter-react-16'; 
Enzyme.configure({adapter: new Adapter()}); 
import renderer from "react-test-renderer";
import { TestScheduler } from "rxjs";

describe ("Home", () => {
  it("renders without crashing", () => {
    const rendered = renderer.create(<Home />).toJSON();
    expect(rendered).toBeTruthy();
    expect(rendered).toMatchSnapshot();
  });
  
  it('should call navigate on pressing login', () => {
    //let handleLoginPressMock = jest.fn(); 
    //let navigationMock = {navigate: jest.fn()}
    const navigationMock = {navigate: jest.fn()}; 

    let homeComponent = renderer.create(<Home 
        navigation={navigationMock}
      />
    ).getInstance();
  
    homeComponent.handleLoginPress(); 
    expect(navigationMock.navigate).toHaveBeenCalledTimes(1); 
  });
  it('should call navigate on pressing GymStats', () => {
    const navigationMock = {navigate: jest.fn()}; 

    let homeComponent = renderer.create(<Home 
        navigation={navigationMock}
      />
    ).getInstance(); 

    homeComponent.handleGymStatsPress(); 
    expect(navigationMock.navigate).toHaveBeenCalledTimes(1); 
  
  });

  it('should call navigate on pressing CodeScanner', () => {
    const navigationMock = {navigate: jest.fn()}; 

    let homeComponent = renderer.create(<Home 
        navigation={navigationMock}
      />
    ).getInstance(); 

    homeComponent.handleCodeScannerPress(); 
    expect(navigationMock.navigate).toHaveBeenCalledTimes(1); 
  
  });

  it('should call Linking on pressing CodeScanner', () => {
    const navigationMock = {navigate: jest.fn()}; 

    let homeComponent = renderer.create(<Home 
        navigation={navigationMock}
      />
    ).getInstance(); 

    homeComponent.handleCodeScannerPress(); 
    expect(navigationMock.navigate).toHaveBeenCalledTimes(1); 
  
  });
  /*
  it("should console.log handle gym stats press", () => {
    handleGymStatsPress = jest.fn(); 
    handleGymStatsPress("handle gym stats press"); 
    expect(handleGymStatsPress.mock.calls[0][0]).toBe("handle gym stats press");
  });
  */
  /*
  it('should navigate onPress button is pressed', () => {
   
    const navigation = {navigate: jest.fn()}; 
    //const spy = jest.spyOn(navigation, 'navigate'); 

    expect(this.props.navigate).toHaveBennCalledTimes(1); 
  })
  */
});




