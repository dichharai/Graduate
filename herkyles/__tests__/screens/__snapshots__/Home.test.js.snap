// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders without crashing 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hitSlop={undefined}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={undefined}
  testID={undefined}
>
  <View
    style={undefined}
  >
    <Image
      resizeMode="contain"
      source={1}
      style={undefined}
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={undefined}
    >
      Herkyles
    </Text>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    collapsable={undefined}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={undefined}
      >
        Gyms
      </Text>
    </View>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    collapsable={undefined}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={undefined}
      >
        Group Fitness Schedule
      </Text>
    </View>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    collapsable={undefined}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={undefined}
      >
        QR Code Scanner
      </Text>
    </View>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    collapsable={undefined}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={undefined}
      >
        Login
      </Text>
    </View>
  </View>
</View>
`;
